pkg/chain/common/gen/Makefile

```diff
@@ -0,0 +1,45 @@
+# If contracts_dir is not defined by a Makefile using this common Makefile default
+# it to a `contracts/` subdirectory under solidity_dir.
+ifndef contracts_dir
+	contracts_dir = $(addsuffix /contracts/,${solidity_dir})
+endif
+
+# Solidity files prefixed with the solidity directory path.
+# Multiple patterns have to be defined to get the files recursively.
+solidity_files := $(wildcard ${contracts_dir}/*.sol) $(wildcard ${contracts_dir}/*/*.sol)
+
+# Go bindings generated for the solidity contracts.
+contract_files = $(addprefix contract/,$(addsuffix .go,${required_contracts}))
+
+# Function to run solidity sources comilation.
+# Expects two arguments:
+# 1 - dependencies mapping for contracts import
+# 2 - solidity source file path
+solc_generate = solc $(1) \
+		--allow-paths ${solidity_dir} \
+		--overwrite \
+		--abi \
+		-o abi $(2)
```

Comment:
Having a specific solc version installed globally is getting problematic. Running `brew install solidity` as in our [`macos-setup.sh`](https://github.com/keep-network/keep-core/blob/b872c35aa405b4e857fba7317aa6bd23137f5a01/scripts/macos-setup.sh#L10-L13) (which is by the way outdated now), installs the most recent version 0.8.15 that does not allow to compile 0.8.9 code. One option is to use a published binary or a tool like [`solc-select`](https://github.com/crytic/solc-select) but binaries are not available for M1 and [it will take some time](https://github.com/ethereum/solidity/issues/12291) until they will be published. It is [no longer that easy](https://stackoverflow.com/questions/3987683/homebrew-install-specific-version-of-formula) to install brew formula from the specific version of a file on GitHub and it requires doing a local tap.

There are two options - we can change `pragma solidity 0.8.9;` to `pragma solidity ^0.8.9;` in all imported projects. I did it locally for the sake of testing this PR on solc 0.8.15 and it works. This is not a good solution though because eventually we want to fix these versions for deployment, tag the specific version, and import the tagged version from Go client.

Another option is to use `hardhat build` to generate ABIs. This feels like a good solution to me because one needs to have `npm` and `yarn` anyway to install the dependencies.

How hard would it be to switch from `solc` to `yarn build` here?

Reply:
We synced with @nkuba about this problem and the idea is to not use `solc` at all. From the Go code perspective, we are interested only in ABI. We do not want to `yarn install` Solidity projects and execute builds. Instead, we want to `wget` packages published to the NPM registry and read ABI from there. It's Solidity project's responsibility to generate ABIs and publish a package.

@nkuba will try to implement the new solution today. By the end of the day, we'll decide if we merge this PR and update the code separately to unblock the rest of the work or if we are able to implement the new solution quickly in this PR.

Reply:
I updated the PR with the new approach.